prow/cmd/deck/main_test.go

```diff
@@ -416,3 +419,145 @@ func TestHelp(t *testing.T) {
 	handleAndCheck()
 	handleAndCheck()
 }
+
+func TestListProwJobs(t *testing.T) {
+	templateJob := &prowapi.ProwJob{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "first",
+			Namespace: "prowjobs",
+			Labels: map[string]string{
+				"foo": "bar",
+			},
+		},
+		Spec: prowapi.ProwJobSpec{
+			Refs: &prowapi.Refs{
+				Org:  "org",
+				Repo: "repo",
+			},
+		},
+	}
+
+	var testCases = []struct {
+		name        string
+		prowJobs    []func(*prowapi.ProwJob) runtime.Object
+		listErr     bool
+		hiddenRepos sets.String
+		hiddenOnly  bool
+		expected    sets.String
+		expectedErr bool
+	}{
+		{
+			name:        "list error results in filter error",
+			listErr:     true,
+			expectedErr: true,
+		},
+		{
+			name: "no hidden repos returns all prowjobs",
+			prowJobs: []func(*prowapi.ProwJob) runtime.Object{
+				func(in *prowapi.ProwJob) runtime.Object { return in },
+			},
+			expected: sets.NewString("first"),
```

Comment:
This pattern is pretty good. If everything we validate in expected gets set in the prowJob generator it will be much easier to read, so `{ in.Name = "first"; return in}` in this scenario